salt/modules/cmdmod.py

```diff
@@ -245,6 +245,10 @@ def _run(cmd,
     '''
     Do the DRY thing and only call subprocess.Popen() once
     '''
+    # Remove whitespace from strings
+    if isinstance(cmd, six.string_types):
+        cmd = cmd.strip()
+
```

Comment:
Can you explain why this would be necessary? We use `shlex.split()` to convert the command to a list to be passed to the `timed_subprocess` instance, and splitting like this already strips leading and trailing whitespace:

```python
>>> import shlex
>>> shlex.split('   foo bar "baz foo"  ')
['foo', 'bar', 'baz foo']
>>>
```

Furthermore, `subprocess.Popen()` doesn't care if there is leading/trailing whitespace:

```python
>>> import subprocess
>>> p = subprocess.Popen('   echo "hello world"  ', shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
>>> p.communicate()
('hello world\n', '')
>>>
```

So, is there a specific reason for this? Something perhaps related to calling `cmd.script` using Powershell?